interface/backend/submission.py

```diff
@@ -42,26 +42,7 @@ def vmck_config(submission):
     return config_dict
 
 
-def handle_submission(request):
-    file = request.FILES['file']
-    log.debug(f'Submission {file.name} received')
-
-    assignment = get_object_or_404(Assignment.objects,
-                                   code=request.GET['assignment_id'])
-    # if not assignment.is_open_for(request.uesr):
-    #     return render('ești bou.html')
-
-    if not assignment:
-        return Http404()
-
-    submission = Submission.objects.create(
-        archive_size=file.size,
-        user=request.user,
-        assignment=assignment,
-    )
-
-    storage.upload(f'{submission.id}.zip', file.read())
-
+def deploy_submission(submission, assignment):
```

Comment:
Yes, extract this, nice. But `deploy` is a concept that makes sense in a cluster (nomad), whereas here we're more like evaluating, so how about `evaluate_submission`? Also you can make it a method of the `submission` object, so it becomes `submission.evaluate()`.